packages/mui-material/src/SpeedDialAction/SpeedDialAction.js

```diff
@@ -173,44 +177,90 @@ const SpeedDialAction = React.forwardRef(function SpeedDialAction(inProps, ref)
 
   const transitionStyle = { transitionDelay: `${delay}ms` };
 
-  const fab = (
-    <SpeedDialActionFab
-      size="small"
-      className={clsx(classes.fab, className)}
-      tabIndex={-1}
-      role="menuitem"
-      ownerState={ownerState}
-      {...FabProps}
-      style={{
-        ...transitionStyle,
-        ...FabProps.style,
-      }}
-    >
-      {icon}
-    </SpeedDialActionFab>
-  );
+  const externalForwardedProps = {
+    slots,
+    slotProps: {
+      fab: FabProps,
+      ...slotProps,
+      tooltip: mergeSlotProps(slotProps.tooltip, {
+        title: tooltipTitle,
+        open: tooltipOpenProp,
+        placement: tooltipPlacement,
+        classes: TooltipClasses,
+      }),
+    },
+  };
```

Comment:
```suggestion
  const externalForwardedProps = {
    slots,
    slotProps: {
      fab: FabProps,
      ...slotProps,
      tooltip: mergeSlotProps(typeof slotProps.tooltip === 'function' ? slotProps.tooltip(ownerState) : slotProps.tooltip, {
        title: tooltipTitle,
        open: tooltipOpenProp,
        placement: tooltipPlacement,
        classes: TooltipClasses,
      }),
    },
  };
  
  const [tooltipOpen, setTooltipOpen] = React.useState(externalForwardedProps.slotProps.tooltip.open);
```

I think you need to move this above `tooltipOpen` state because `slotProps.tooltip` can be a function (the current code will break for function).

Reply:
updated here https://github.com/mui/material-ui/pull/45065/commits/7b8a99dbad4e6f8dca1c021df285941595f056ff